src/app/launcher/shared/telemetry.decorator.ts

```diff
@@ -26,8 +26,13 @@ export function broadcast(event: string, properties: any): MethodDecorator {
             }
 
             const broadcast: Broadcaster = injectorInstance.get(Broadcaster);
-            const values = new PropertiesGetter(this).mapKeys(_.cloneDeep(properties));
+            let values = new PropertiesGetter(this).mapKeys(_.cloneDeep(properties));
 
+            if (event === 'stepIndicatorClicked') {
+                values = {
+                    step: args[0] ? args[0] : ''
+                };
+            }
```

Comment:
this seems very specific to one case, why would this be needed?

Reply:
For this event here, i get the data from the original method's argument. so I had to add the specific thing here.